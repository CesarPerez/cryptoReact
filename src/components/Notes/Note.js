import * as React from 'react';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

const Note = props => {
  const { text, handlerRemove, id } = props;
  return (
    <Card sx={{ minWidth: 150 }}>
      <CardContent>
        <Typography sx={{ mb: 1.5 }} color="text.secondary">
          {text}
        </Typography>
      </CardContent>
      <CardActions>
        <Button size="small" onClick={handlerRemove} value={id}>
          Delete
        </Button>
      </CardActions>
    </Card>
  );
};

export default Note;
