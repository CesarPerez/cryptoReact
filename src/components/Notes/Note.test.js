import renderer from 'react-test-renderer'; // ES6
import Note from './Note';

test('Should match the snapshot', () => {
  const handlerRemoveMock = jest.fn();
  const dummyText = 'dummyText';
  const dummyId = 1;
  const tree = renderer.create(<Note text={dummyText} handlerRemove={handlerRemoveMock} id={dummyId} />).toJSON();
  expect(tree).toMatchSnapshot();
});
