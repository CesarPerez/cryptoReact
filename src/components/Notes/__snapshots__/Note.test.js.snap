// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-6eycvs-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-9eg9l8-MuiTypography-root"
    >
      dummyText
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value={1}
    >
      Delete
    </button>
  </div>
</div>
`;
